llvm/lib/Target/NVPTX/NVPTXTargetTransformInfo.cpp

```diff
@@ -458,7 +459,7 @@ handleSpaceCheckIntrinsics(InstCombiner &IC, IntrinsicInst &II) {
       if (auto *ASCO = dyn_cast<AddrSpaceCastOperator>(Op0))
         AS = ASCO->getOperand(0)->getType()->getPointerAddressSpace();
```

Comment:
Do we still need this attempt to peek through the ASC then? If we do manage to see original AS, so would the `infer-address-spaces` pass, and it would have folded the intrinsic into a constant.

Reply:
Where do we run AS inference pass in the pipeline? If it's early, then I think it would make sense to keep isspacep folding in one place only, and AS inference pass looks like a better place for that.

Reply:
Looks like it is only run in `llc`. I see AMDGPU runs this pass in `opt` so we should maybe consider doing the same, but I'm not sure how we'd make that decision.

Reply:
OK. Add a TODO comment there for now and we'll handle it separately.